client/src/pages/Login.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useState } from "react";
+import { TextInputField } from "./../components/Components";
+import { Container, Grid, Typography, Button } from "@material-ui/core";
+import { IdentityToken, emptyIdentityToken } from "./../interfaces";
+import { extractIdentityToken } from "./../utils";
+import { TITLE_TEXT, microapps, isWeb, isDebug } from "./../constants/index";
+import Logo from "./../img/Logo.png";
+import "./../css/Login.css";
+declare const window: any;
+
+function Login() {
+  const logoSpinnerSpeed = 3;
+
+  // Flag for us to manually run login on mobile
+  const [loginError, setLoginError] = useState(false);
+  // User Identity
+  const [identity, setIdentity] = useState<IdentityToken>(emptyIdentityToken());
+
+  const updateUser = (text: string) => {
+    const newIdentity = emptyIdentityToken();
+    newIdentity.sub = text;
+    setIdentity(newIdentity);
+  };
+
+  const login = () => {
+    if (isWeb) {
+      return;
+    }
+    const request = { nonce: "Don't Hack me please" };
+    microapps
+      .getIdentity(request)
+      .then((response: any) => {
+        setIdentity(extractIdentityToken(response));
+      })
+      .catch((error: any) => {
+        setLoginError(true);
+      });
+  };
+
+  useEffect(() => {
+    if (identity.sub != "EMPTY" && identity.sub != "") {
+      window.location.href = "/home";
+    }
+  }, [identity]);
+
+  useEffect(() => {
+    // Initial login if on microapp
+    login();
+  }, []);
```

Comment:
sorry if i have asked this before, is this also based on identity? if so do we want to add [identity] here?

Reply:
answered in #74